src/core/src/package_managers/PackageManager.py

```diff
@@ -321,37 +321,43 @@ def set_package_manager_setting(self, setting_key, setting_value=""):
     # endregion
 
     # region auto OS updates
+    @abstractmethod
     def disable_auto_os_update(self):
         pass
 
-    def log_default_auto_os_updates(self):
+    def add_image_default_patch_mode_backup(self):
         """ Records the default system settings for auto OS updates within patch extension artifacts for future reference.
         We only log the default setting a VM comes with, any subsequent updates will not be recorded"""
-        if not self.default_auto_os_update_log_exists():
-            current_auto_os_update_settings = self.get_current_auto_os_update_settings()
-            if current_auto_os_update_settings is not None:
+        try:
+            if not self.image_default_patch_mode_backup_exists():
+                current_auto_os_update_settings = self.get_current_auto_os_update_settings()
                 self.composite_logger.log_debug("Logging default system settings for auto OS updates. [Settings={0}] [Log file path= {1}]"
-                                                .format(str(current_auto_os_update_settings), self.default_auto_os_update_log_path))
-                self.env_layer.file_system.write_with_retry(self.default_auto_os_update_log_path, '[{0}]'.format(current_auto_os_update_settings), mode='w+')
+                                                .format(str(current_auto_os_update_settings), self.image_default_patch_mode_backup_path))
+                self.env_layer.file_system.write_with_retry(self.image_default_patch_mode_backup_path, '[{0}]'.format(current_auto_os_update_settings), mode='w+')
+        except Exception as error:
+            error_message = "Exception during fetching and logging default auto update settings on the machine. [Exception={0}]".format(repr(error))
+            self.composite_logger.log_error(error_message)
+            raise Exception
 
-    def default_auto_os_update_log_exists(self):
+    def image_default_patch_mode_backup_exists(self):
         """ Checks whether default auto OS updates have been recorded earlier within patch extension artifacts """
         self.composite_logger.log_debug("Reading auto OS updates log...")
-        if not os.path.exists(self.default_auto_os_update_log_path) or not os.path.isfile(self.default_auto_os_update_log_path):
+        if not os.path.exists(self.image_default_patch_mode_backup_path) or not os.path.isfile(self.image_default_patch_mode_backup_path):
             self.composite_logger.log_debug("Default system settings for auto OS updates aren't recorded in the extension")
```

Comment:
Check if anything needs to be done with the config folder in an extension upgrade scenario.

Reply:
Please ensure that the extension upgrade story is covered to retain the image_default configuration.

Reply:
Copying image_default backup, CoreState.json and ExtState.json in update handler action. Do we also need to copy the config settings file?

Reply:
The agent is responsible for maintaining the settings file. Is there a reason you think we should copy it?

Reply:
The only one I can think of is, if we are debugging any previous extension runs and want to find out what config settings were sent in for that